crates/net/rpc/src/admin/mod.rs

```diff
@@ -0,0 +1,55 @@
+use jsonrpsee::core::RpcResult;
+use reth_network::{peers::PeerKind, DisconnectReason, NetworkHandle};
+use reth_primitives::NodeRecord;
+use reth_rpc_api::AdminApiServer;
+
+struct AdminApi {
+    /// An interface to interact with the network
+    network: NetworkHandle,
+}
+
+impl AdminApiServer for AdminApi {
+    fn add_peer(&self, record: String) -> RpcResult<bool> {
```

Comment:
Since `NodeRecord`'s deserialize is jus `FromStr` we can change the argument from String to `NodeRecord` directly, makes everything cleaner.

Reply:
But if the record can"t be deserialized into `NodeRecord`, the methods will return an Err(...) instead of Ok(false). Is that ok?

Reply:
then the argument is wrong, and the call fails, which I think should be the correct behavior?